test_plans/execution_model.asciidoc

```diff
@@ -0,0 +1,119 @@
+
+== Testing scope
+
+=== Backend coverage
+
+All the tests described below are performed for any SYCL implementation.
+
+=== Device coverage
+
+All of the tests described below are performed only on the default device that
+is selected on the CTS command line.
+
+== Tests
+
+=== Execution order of three command groups submitted to the same queue
+
+Create 2 buffers for large arrays. Submit 2 commands that have some long loops and then fill these buffers with expected values via accessors in loop.
+Submit command that copy content of these buffers to result buffer.
+
+[source,c++]
+----
+queue.submit([&](handler& cgh) {
+  sycl::accessor<int, 1> acc1(buffer1, cgh);
+  cgh.single_task([=] {
+    // long loop
+    for (int i = SIZE; i >= 0; i--)
+      acc1[i] = i;
+  });
+});
+queue.submit([&](handler& cgh) {
+  sycl::accessor<int, 1> acc2(buffer2, cgh);
+  cgh.single_task([=] {
+    // long loop
+    for (int i = SIZE; i >= 0; i--)
+      acc2[i] = i;
+  });
+});
+queue.submit([&](handler& cgh) {
+  sycl::accessor<int, 1> acc1(buffer1, cgh);
+  sycl::accessor<int, 1> acc2(buffer2, cgh);
+  sycl::accessor<int, 1> res_acc1(res_buffer1, cgh);
+  sycl::accessor<int, 1> res_acc2(res_buffer2, cgh);
+  cgh.parallel_for(sycl::range<1>(SIZE),
+                   [=](id<1> index) {
+                     res_acc1[index] = acc1[index];
+                     res_acc2[index] = acc2[index];
+                  });
+});
+----
+
+Check that all elements in result buffer are equal to expected values which means that requisites for third command were satisfied before it started to execute.
+
+=== Execution order of three command groups submitted to the different queues
+
+Same as previous tests but each command is run on different queue with diffrent context.
+
+=== Requirements on overlapping vs non-overlapping sub-buffer
+
+This check is skipped if device doesn't support `aspect::usm_atomic_shared_allocations`.
+
+Create buffer on array, and two non-overlapping sub-buffers
+
+[source,c++]
+----
+buffer<int, 1> sub_buf1(buffer, 0, 5);
+buffer<int, 1> sub_buf2(buffer, 10, 5);
+int *pflag = sycl::malloc_shared<int>(1, queue);
+queue.submit([&](handler& cgh) {
+  sycl::accessor<int, 1> acc1(sub_buf1, cgh);
+  cgh.single_task([=] {
+    // long loop
+    for (int i = SIZE; i >= 0; i--)
+      acc1[i] = i;
+    *pflag = 1;
+  });
+});
+queue.submit([&](handler& cgh) {
+  sycl::accessor<int, 1> acc2(sub_buf2, cgh);
+  cgh.single_task([=] {
+    for (int i = SIZE; i >= 0; i--)
+      acc2[i] = i;
+    *pflag = 2;
+  });
+});
+queue.submit([&](handler& cgh) {
+  sycl::accessor<int, 1> acc(buffer, cgh);
+  sycl::accessor<int, 1> res_acc(res_buffer, cgh);
+  copy(acc1, res_acc);
+});
+----
+
+Check that *pflag is equal to 1, that means first two commands executes independently, and check that all elements in result buffer are equal to expected values which means that requisites for third command were satisfied before it started to execute.
```

Comment:
For me we can have either `*pflag` equals to 1 or 2 when the copy happens, but how to test it with this `cgh.copy()`. I think you need a normal kernel instead to read it and write it for example in another buffer.